ITISFoundation/osparc-simcore
# Is2911/fix delete project

## What do these changes do?

New project deletion takes the following steps:
1. marks project row as invisible (next PR will mark it as ``trashed``)
2. stops dynamic services
3. stops computational services
4. calls to delete data (next PR will mark it as ``trashed``)
5. deletes project (row database). 

The key difference is that now the project is deleted last which keeps all the access-rights information necessary to perform the deletion in storage. 

NEXT: A follow up to this case will be a trash for projects ITISFoundation/osparc-issues#468. There we will also extend delete workflows and integration tests.

#### Misc

- ♻️ cleanup ``test_garbage_collector`` as well as some logs in gc plugin.
- ♻️ split ``projects_handler`` since >500 lines
- add storage info on show endpoint makefile recipe
- fixed deprecation warning:
```
 DeprecationWarning: The 'asyncio_mode' is 'legacy', switching to 'auto' for the sake of pytest-aiohttp backward compatibility. Please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto' in pytest configuration file.
    config.issue_config_time_warning(LEGACY_MODE, stacklevel=2)
```

## Related issue/s

- Fixes #2911 
- Fixes #2754
- closes PR #2777

## How to test

#### Manual testing 

NOTE: will provide an internal entrypoint for testing on Friday

- Manual test workflow described in #2911
- Manual test workflow described in #2754
- Manual test with @odeimaiz 
   -  create project 1, upload file, close
   -  create new project 2, add file-picker and select previous file, close file
   -  delete project 2 -> should not delete upload file because it is in project 1

#### Automatic CI
```
pytest  tests/**/test_*delete*.py
```
will run
```
<Module tests/unit/with_dbs/02/test_projects_delete.py>
  <Function test_delete_project[UserRole.ANONYMOUS-expected0]>
  <Function test_delete_project[UserRole.GUEST-expected1]>
  <Function test_delete_project[UserRole.USER-expected2]>
  <Function test_delete_project[UserRole.TESTER-expected3]>
  <Function test_delete_multiple_opened_project_forbidden[UserRole.ANONYMOUS-HTTPUnauthorized-HTTPUnauthorized]>
  <Function test_delete_multiple_opened_project_forbidden[UserRole.GUEST-HTTPOk-HTTPForbidden]>
  <Function test_delete_multiple_opened_project_forbidden[UserRole.USER-HTTPOk-HTTPForbidden]>
  <Function test_delete_multiple_opened_project_forbidden[UserRole.TESTER-HTTPOk-HTTPForbidden]>
```
